cider/exec/template/StringOpsIR.cpp

```diff
@@ -111,6 +111,27 @@ extern "C" RUNTIME_EXPORT int32_t lower_encoded(int32_t string_id,
 #endif
 }
 
+#define DEF_APPLY_NUMERIC_STRING_OPS(value_type, value_name)                             \
+  extern "C" RUNTIME_EXPORT ALWAYS_INLINE value_type                                     \
+      apply_numeric_string_ops_##value_name(                                             \
+          const char* str_ptr, const int32_t str_len, const int64_t string_ops_handle) { \
+    std::string raw_str(str_ptr, str_len);                                               \
+    auto string_ops =                                                                    \
+        reinterpret_cast<const StringOps_Namespace::StringOps*>(string_ops_handle);      \
+    const auto result_datum = string_ops->numericEval(raw_str);                          \
+    return result_datum.value_name##val;                                                 \
```

Comment:
can we skip cast_type branch check inside` numericEval` method? eg.separate and call different cast function from here string_to_int, string_to_date...

Reply:
I think we can. It is a good place to switch type and cast string to there different types